src/pages/TravelTipsPage/TravelTipsAllCards.jsx

```diff
@@ -0,0 +1,74 @@
+import TravelTipsBtn from '../../components/common/Buttons/TravelTipsButtons/TravelTipsBtn';
+import SearchPanelTravelTips from '../../components/common/SearchPanels/SearchPanelTravelTips';
+import TravelTipsCard from '../../components/common/TravelTipsCard/TravelTipsCard';
+import tipsData from '../../data/tipsData';
+import { useState, memo } from 'react';
+
+const TravelTipsAllCards = () => {
+  const [activeButtons, setActiveButtons] = useState([]);
+
+  const travelTipsButtons = [
+    { text: 'Packing' },
+    { text: 'Transportation' },
+    { text: 'Accommodation' },
+    { text: 'Local Cuisine' },
+    { text: 'Cheap Trip' },
+    { text: 'All Tips' },
+  ];
+
+  const handleButtonClick = buttonName => {
+    setActiveButtons(prevActiveButtons => {
+      if (buttonName === 'All Tips') {
+        return prevActiveButtons.includes(buttonName)
+          ? []
+          : travelTipsButtons.map(button => button.text);
+      }
+
+      const isActive = prevActiveButtons.includes(buttonName);
+      if (isActive) {
+        const updatedButtons = prevActiveButtons.filter(
+          name => name !== buttonName,
+        );
+        return updatedButtons.length === 0 ? [] : updatedButtons;
```

Comment:
```suggestion
        return updatedButtons;
```

You don't need this check, `filter` is not mutative function and it will return an empty array instead of throwing an error